provider/cmd/pulumi-gen-awsx/ecs.go

```diff
@@ -0,0 +1,248 @@
+		TypeSpec: schema.TypeSpec{
+			Ref:   "#/types/awsx:iam:DefaultRoleWithPolicy",
+			Plain: true,
+		},
+	}
+
+	return schema.ResourceSpec{
+		IsComponent: true,
+		ObjectTypeSpec: schema.ObjectTypeSpec{
+			Description: "Create a TaskDefinition resource with the given unique name, arguments, and options.\nCreates required log-group and task & execution roles.\nPresents required Service load balancers if target group included in port mappings.",
+			Properties: map[string]schema.PropertySpec{
+				"taskDefinition": {
+					Description: "Underlying ECS Task Definition resource",
+					TypeSpec: schema.TypeSpec{
+						Ref: awsRef("#/resources/aws:ecs%2FtaskDefinition:TaskDefinition"),
+					},
+				},
+				"logGroup": {
+					Description: "Auto-created Log Group resource for use by containers.",
+					TypeSpec: schema.TypeSpec{
+						Ref: awsRef("#/resources/aws:cloudwatch%2FlogGroup:LogGroup"),
+					},
+				},
+				"taskRole": {
+					Description: "Auto-created IAM role that allows your Amazon ECS container task to make calls to other AWS services.",
+					TypeSpec: schema.TypeSpec{
+						Ref: awsRef("#/resources/aws:iam%2Frole:Role"),
+					},
+				},
+				"executionRole": {
+					Description: "Auto-created IAM task execution role that the Amazon ECS container agent and the Docker daemon can assume.",
+					TypeSpec: schema.TypeSpec{
+						Ref: awsRef("#/resources/aws:iam%2Frole:Role"),
+					},
+				},
+				"loadBalancers": {
+					Description: "Computed load balancers from target groups specified of container port mappings.",
+					TypeSpec: schema.TypeSpec{
+						Type: "array",
+						Items: &schema.TypeSpec{
+							Ref: awsRef("#/types/aws:ecs%2FServiceLoadBalancer:ServiceLoadBalancer"),
+						},
+					},
+				},
+			},
+			Required: []string{"taskDefinition", "loadBalancers"},
+		},
+		InputProperties: inputProperties,
+	}
+}
+
+func containerDefinitionTypes(awsNativeSpec schema.PackageSpec) map[string]schema.ComplexTypeSpec {
+	names := []string{
+		"TaskDefinitionContainerDefinition",
+		"TaskDefinitionContainerDependency",
+		"TaskDefinitionKeyValuePair",
+		"TaskDefinitionEnvironmentFile",
+		"TaskDefinitionHostEntry",
+		"TaskDefinitionFirelensConfiguration",
+		"TaskDefinitionHealthCheck",
+		"TaskDefinitionLinuxParameters",
+		"TaskDefinitionLogConfiguration",
+		"TaskDefinitionMountPoint",
+		"TaskDefinitionPortMapping",
+		"TaskDefinitionRepositoryCredentials",
+		"TaskDefinitionResourceRequirement",
+		"TaskDefinitionSecret",
+		"TaskDefinitionSystemControl",
+		"TaskDefinitionUlimit",
+		"TaskDefinitionVolumeFrom",
+		"TaskDefinitionKernelCapabilities",
+		"TaskDefinitionDevice",
+		"TaskDefinitionTmpfs",
+		"TaskDefinitionSecret",
+	}
+	types := map[string]schema.ComplexTypeSpec{}
+	for _, name := range names {
+		types["awsx:ecsx:"+name] = renameComplexRefs(awsNativeSpec.Types["aws-native:ecs:"+name], "aws-native:ecs:", "awsx:ecsx:")
```

Comment:
Definitely worth documenting the reasoning and the mechanics of using the aws-native types here.